app/models/user.rb

```diff
@@ -69,8 +69,12 @@ def send_two_factor_authentication_code(_code)
     # See https://github.com/18F/identity-idp/pull/452 for more details.
   end
 
+  def confirmed?
+    email_addresses.where.not(confirmed_at: nil).any?
+  end
+
   def confirmation_period_expired?
-    confirmation_sent_at.present? && confirmation_sent_at.utc <= self.class.confirm_within.ago
+    super
```

Comment:
Since there is no `confirmation_period_expired?` method in ApplicationRecord, what will happen when `super` is called?

Reply:
This method is patched in by Devise (https://github.com/plataformatec/devise/blob/v4.6.2/lib/devise/models/confirmable.rb#L232). We overrode it because it is protected and we call it in other parts of the code. This change changes it so it will use the latest Devise code even after we update the gem.

Reply:
Oh, I missed the `devise(...)` code block at line 6 of user.rb.  Thanks!